<launch>


  <!-- IK -->
  <!-- == -->

  <!-- Setup IK -->
  <node pkg="ik_ros" name="ik_setup" type="ik_setup_node.py" args="trac_ik" output="screen">
    <param name="parent_frame_id" type="str" value="rpbi/kuka_lwr/panda_link0"/>
    <param name="child_frame_id" type="str" value="interpol_target"/>
    <param name="start_on_init" type="bool" value="true"/>
    <param name="init_brx" type="double" value="10.0"/>
    <param name="init_bry" type="double" value="10.0"/>
    <param name="init_brz" type="double" value="10.0"/>
    <remap from="ik/setup/trac_ik/qinit" to="rpbi/kuka_lwr/joint_states"/>
  </node>



  <!-- IK solver interface -->
  <node pkg="ik_ros" name="ik_node" type="ik_solver_node.py" args="trac_ik" output="screen">
    <param name="urdf_filename" type="string" value="{franka_panda_description}/robots/panda_arm.urdf"/>
    <param name="tip_link" type="string" value="panda_link0"/>
    <param name="base_link" type="string" value="panda_link8"/>
    <param name="start_on_init" type="bool" value="true"/>
    <remap from="joint_states/target" to="rpbi/kuka_lwr/joint_states/target"/>
  </node>



  <!-- Simulation -->
  <!-- ========== -->

  <!-- ROS-Pybullet interface node simulator -->
  <node pkg="ros_pybullet_interface" name="ros_pybullet_interface_node" type="ros_pybullet_interface_node.py">
    <rosparam param="config" file="$(find ros_quat_ds)/configs/interpolation/config.yaml"/>
  </node>



  <!-- Static transforms -->
  <!-- corresponds to green sphere -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="visual_wpt1_tf"
	args="-0.4 -0.0 0.1 0 0 0 rpbi/world rpbi/visual_wpt1"/>

  <!-- corresponds to blue sphere -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="visual_wpt2_tf"
	args="-0.5 0.2 0.3 0 0 0 rpbi/world rpbi/visual_wpt2"/>

  <!-- corresponds to red sphere -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="visual_wpt3_tf"
	args="-0.15, 0.4, 0.2 0 0 0 rpbi/world rpbi/visual_wpt3"/>


  <!-- Interpolation -->
  <!-- ============= -->
  <node  pkg="rpbi_utils" name="interpolation"  type="interpolation_node.py" output="screen">
    <!-- Traj config (required) -->
    <param name="traj_config" value="$(find ros_quat_ds)/configs/interpolation/traj_lwr.yaml" type="str"/>
    <param name="consuming_freq" value="100.0" type="double"/>

    <remap from="ros_pybullet_interface/waypt_traj" to="kuka_lwr/ros_pybullet_interface/end_effector/waypt_traj"/>


 </node>

  <!-- Run example -->
  <!-- =========== -->

  <node pkg="ros_quat_ds" name="run" type="run_interpolation.py" output="screen">
    <param name="topic_name_4_waypoints" value="kuka_lwr/ros_pybullet_interface/end_effector/waypt_traj" type="str"/>
  </node>

</launch>
